test/e2e/operator_test.go

```diff
@@ -177,17 +177,48 @@ func TestDefaultDNSSteadyConditions(t *testing.T) {
 }
 
 func TestCoreDNSImageUpgrade(t *testing.T) {
-	// TODO: Revise this test and remove skip.
-	// See https://bugzilla.redhat.com/show_bug.cgi?id=1912522
-	t.Skip("skipping TestCoreDNSImageUpgrade as this test is fundamentally broken")
 	cl, err := getClient()
 	if err != nil {
 		t.Fatal(err)
 	}
+	// Override the cluster version to temporarily disable CVO updates to the DNS operator deployment.
+	deplOverride := configv1.ComponentOverride{
+		Kind:      "Deployment",
+		Group:     "apps/v1",
+		Namespace: "openshift-dns-operator",
+		Name:      "dns-operator",
+		Unmanaged: true,
+	}
+
+	clusterVersion := &configv1.ClusterVersion{}
+	cvName := types.NamespacedName{Namespace: "default", Name: "version"}
+
+	if err := cl.Get(context.TODO(), cvName, clusterVersion); err != nil {
+		t.Logf("failed to get cluster version %s/%s: %v", cvName.Namespace, cvName.Name, err)
+	}
+
+	clusterVersion.Spec.Overrides = []configv1.ComponentOverride{deplOverride}
+
+	if err := cl.Update(context.TODO(), clusterVersion); err != nil {
+		t.Logf("failed to update cluster version %s/%s: %v", cvName.Namespace, cvName.Name, err)
+	}
+
+	// Remove cluster version override once the test returns.
+	defer func() {
+		if err := cl.Get(context.TODO(), cvName, clusterVersion); err != nil {
+			t.Logf("failed to get cluster version %s/%s: %v", cvName.Namespace, cvName.Name, err)
```

Comment:
I really think we should fail harder.  Can the next test continue if this fails?

Reply:
Arguably yes, the following tests _could_ run, but it's probably best that we ensure cluster version is in a clean state first.

Reply:
fixed.

Reply:
Also made the prior cluster version override API calls hard fail the test if they fail. Thanks!